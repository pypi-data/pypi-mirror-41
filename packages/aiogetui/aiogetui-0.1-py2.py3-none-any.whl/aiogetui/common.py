import uuid
from template import Template


class Message:

    def __init__(self,
                 client_id: str,
                 template: Template,
                 is_offline: bool = False,
                 offline_expire_time: int = 0,
                 message_id: str = None,
                 push_info: dict = None):
        """
        :param client_id: 客户端的 id
        :param template: 消息内容模板
        :param is_offline: 是否离线发送
        :param offline_expire_time: 离线消息的有效期(单位: ms)
        :param message_id: 长度为 10~32
        """
        self.client_id = client_id
        self.template = template
        self.is_offline = is_offline
        self.offline_expire_time = offline_expire_time

        if message_id is None:
            self.message_id = uuid.uuid4().hex  # 长度限制为 32
        else:
            self.message_id = message_id

        # 当手机为 ios，并且为离线的时候, 可是通过 push_info 发送消息
        self.push_info = push_info

        self.duration = 0

    def to_params(self, app_key: str):
        raise NotImplementedError


class ToSingleMessage(Message):
    """单独推送的消息
    """

    def to_params(self, app_key: str):
        params = {
            'cid': self.client_id,
            'requestid': self.message_id,
            'message': {
                'appkey': app_key,
                'is_offline': self.is_offline,
                'offline_expire_time': self.offline_expire_time,
                'msgtype': self.template.type,
            },
            self.template.type: self.template.to_dict(),
        }
        if self.push_info:
            params['push_info'] = self.push_info
        return params


class PushResult:
    OK = 'ok'
    INVALID_CLIENT_ID = 'no_user'
    HTTP_REQUEST_FAILED = 'http_request_failed'

    def __init__(self,
                 result: str,
                 description: str=None,
                 task_id: str=None,
                 status: str=None):
        self.result = result
        self.description = description
        self.task_id = task_id
        self.status = status

    def __str__(self):
        return f'PushResult<' \
               f'result: {self.result}, ' \
               f'description: {self.description}, ' \
               f'task_id: {self.task_id}, ' \
               f'status: {self.status}, ' \
               f'>'

    @property
    def is_successful(self):
        return self.status == self.OK

    @property
    def is_client_id_invalid(self):
        return self.status == self.INVALID_CLIENT_ID
