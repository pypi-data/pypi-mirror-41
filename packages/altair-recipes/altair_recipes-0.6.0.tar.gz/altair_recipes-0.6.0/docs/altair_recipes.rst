altair\_recipes package
=======================

Module contents
---------------

.. automodule:: altair_recipes
    :members:
    :undoc-members:
    :show-inheritance:
