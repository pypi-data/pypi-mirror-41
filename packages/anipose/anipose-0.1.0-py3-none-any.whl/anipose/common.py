import cv2
import re
import os, os.path
from collections import deque
from glob import glob
import skvideo.io
from subprocess import check_output
import toml

def atoi(text):
    return int(text) if text.isdigit() else text

def natural_keys(text):
    '''
    alist.sort(key=natural_keys) sorts in human order
    http://nedbatchelder.com/blog/200712/human_sorting.html
    '''
    return [ atoi(c) for c in re.split('(\d+)', text) ]

def wc(filename):
    out = check_output(["wc", "-l", filename])
    num = out.decode('utf8').split(' ')[0]
    return int(num)

def get_data_length(fname):
    return wc(fname) - 1

def get_video_params(fname):
    cap = cv2.VideoCapture(fname)

    params = dict()
    params['width'] = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
    params['height'] = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
    params['fps'] = cap.get(cv2.CAP_PROP_FPS)

    cap.release()

    return params

def get_folders(path):
    folders = next(os.walk(path))[1]
    return sorted(folders)


def true_basename(fname):
    basename = os.path.basename(fname)
    basename = os.path.splitext(basename)[0]
    return basename


def get_cam_name(config, fname):
    basename = true_basename(fname)

    cam_regex = config['triangulation']['cam_regex']
    match = re.search(cam_regex, basename)

    if not match:
        return None
    else:
        name = match.groups()[0]
        return name.strip()

def get_video_name(config, fname):
    basename = true_basename(fname)

    cam_regex = config['triangulation']['cam_regex']
    vidname = re.sub(cam_regex, '', basename)
    return vidname.strip()

def get_duration(vidname):
    metadata = skvideo.io.ffprobe(vidname)
    duration = float(metadata['video']['@duration'])
    return duration

def get_nframes(vidname):
    metadata = skvideo.io.ffprobe(vidname)
    length = int(metadata['video']['@nb_frames'])
    return length

def full_path(path):
    path_user = os.path.expanduser(path)
    path_full = os.path.abspath(path_user)
    path_norm = os.path.normpath(path_full)
    return path_norm


def process_all(config, process_session, **args):

    pipeline_prefix = config['path']
    nesting = config['nesting']

    output = dict()

    if nesting == 0:
        output[()] = process_session(config, pipeline_prefix, **args)
        return output

    folders = get_folders(pipeline_prefix)
    level = 1

    q = deque()

    next_folders = [ (os.path.join(pipeline_prefix, folder),
                      (folder,),
                      level)
                     for folder in folders ]
    q.extend(next_folders)

    while len(q) != 0:
        path, past_folders, level = q.pop()
        if level == nesting:
            output[past_folders] = process_session(config, path, **args)
        elif level > nesting:
            continue
        elif level < nesting:
            folders = get_folders(path)
            next_folders = [ (os.path.join(path, folder),
                              past_folders + (folder,),
                              level+1)
                             for folder in folders ]
            q.extend(next_folders)

    return output

def make_process_fun(process_session, **args):
    def fun(config):
        return process_all(config, process_session, **args)
    return fun

def find_calibration_folder(config, session_path):
    pipeline_calibration_videos = config['pipeline_calibration_videos']
    nesting = config['nesting']

    level = nesting
    curpath = session_path

    while level >= 0:
        checkpath = os.path.join(curpath, pipeline_calibration_videos)
        print(checkpath)
        videos = glob(os.path.join(checkpath, '*.avi'))
        if len(videos) > 0:
            return curpath

        curpath = os.path.dirname(curpath)
        level -= 1

def load_intrinsics(folder, cam_names):
    intrinsics = {}
    for cname in cam_names:
        fname = os.path.join(folder, 'intrinsics_{}.toml'.format(cname))
        intrinsics[cname] = toml.load(fname)
    return intrinsics

def load_extrinsics(folder):
    extrinsics = toml.load(os.path.join(folder, 'extrinsics.toml'))
    extrinsics_out = dict()
    for k, v in extrinsics.items():
        new_k = tuple(k.split('_'))
        extrinsics_out[new_k] = v
    return extrinsics_out
