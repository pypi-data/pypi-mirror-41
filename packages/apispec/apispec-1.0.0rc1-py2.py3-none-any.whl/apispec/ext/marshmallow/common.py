# -*- coding: utf-8 -*-
"""Utilities to get schema instances/classes"""

import copy
import warnings
from collections import namedtuple

import marshmallow


MODIFIERS = ['only', 'exclude', 'load_only', 'dump_only', 'partial']


def resolve_schema_instance(schema):
    """Return schema instance for given schema (instance or class)

    :param type|Schema|str schema: instance, class or class name of marshmallow.Schema
    :return: schema instance of given schema (instance or class)
    """
    if isinstance(schema, type) and issubclass(schema, marshmallow.Schema):
        return schema()
    if isinstance(schema, marshmallow.Schema):
        return schema
    return marshmallow.class_registry.get_class(schema)()


def resolve_schema_cls(schema):
    """Return schema class for given schema (instance or class)

    :param type|Schema|str: instance, class or class name of marshmallow.Schema
    :return: schema class of given schema (instance or class)
    """
    if isinstance(schema, type) and issubclass(schema, marshmallow.Schema):
        return schema
    if isinstance(schema, marshmallow.Schema):
        return type(schema)
    return marshmallow.class_registry.get_class(schema)


def get_fields(schema):
    """Return fields from schema"""
    if hasattr(schema, 'fields'):
        return schema.fields
    elif hasattr(schema, '_declared_fields'):
        return copy.deepcopy(schema._declared_fields)
    raise ValueError("{0!r} doesn't have either `fields` or `_declared_fields`.".format(schema))


def make_schema_key(schema):
    if not isinstance(schema, marshmallow.Schema):
        raise TypeError('can only make a schema key based on a Schema instance.')
    modifiers = []
    for modifier in MODIFIERS:
        attribute = getattr(schema, modifier)
        try:
            hash(attribute)
        except TypeError:
            attribute = tuple(attribute)
        modifiers.append(attribute)
    return SchemaKey(schema.__class__, *modifiers)


SchemaKey = namedtuple('SchemaKey', ['SchemaClass'] + MODIFIERS)


def get_unique_schema_name(components, name, counter=0):
    """Function to generate a unique name based on the provided name and names
    already in the spec.  Will append a number to the name to make it unique if
    the name is already in the spec.

    :param Components components: instance of the components of the spec
    :param string name: the name to use as a basis for the unique name
    :param int counter: the counter of the number of recursions
    :return: the unique name
    """
    if name not in components._schemas:
        return name
    if not counter:  # first time time through recursion
        warnings.warn(
            'Multiple schemas resolved to the name {} - name has been modified '
            'either manually add each of the schemas with a different name or '
            'provide custom schema_name_resolver'.format(name),
            UserWarning,
        )
    else:  # subsequent recursions
        name = name[:-len(str(counter))]
    counter += 1
    return get_unique_schema_name(components, name + str(counter), counter)
