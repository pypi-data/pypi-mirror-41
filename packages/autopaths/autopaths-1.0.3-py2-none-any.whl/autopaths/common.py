# Built-in modules #
import os, hashlib

################################################################################
def prepend_to_file(path, data, bufsize=1<<15):
    """TODO:
    * Add a random string to the backup file.
    * Restore permissions after copy.
    """
    # Backup the file #
    backupname = path + os.extsep + 'bak'
    # Remove previous backup if it exists #
    try: os.unlink(backupname)
    except OSError: pass
    os.rename(path, backupname)
    # Open input/output files, note: outputfile's permissions lost #
    with open(backupname) as inputfile:
        with open(path, 'w') as outputfile:
            outputfile.write(data)
            buf = inputfile.read(bufsize)
            while buf:
                outputfile.write(buf)
                buf = inputfile.read(bufsize)
    # Remove backup on success #
    os.remove(backupname)

################################################################################
def append_to_file(path, data):
    with open(path, "a") as handle:
        handle.write(data)

################################################################################
def md5sum(file_path, blocksize=65536):
    """Compute the md5 of a file. Pretty fast."""
    md5 = hashlib.md5()
    with open(file_path, "rb") as f:
        for block in iter(lambda: f.read(blocksize), ""):
            md5.update(block)
    return md5.hexdigest()
