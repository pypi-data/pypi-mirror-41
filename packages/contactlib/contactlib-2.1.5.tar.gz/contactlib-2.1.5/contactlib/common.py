import os
import platform
import sys
import numpy as np

from Bio.PDB.DSSP import *
from Bio.PDB.PDBParser import *
from Bio.PDB.Polypeptide import *
from scipy.spatial.distance import *

from contactlib.data_manger import asset_path
from contactlib.timer import TimeIt


def convertPDB(pdbfn):
  if platform.system() == "Darwin":
    cmd = asset_path("dssp-2.0.4-macOS")
  elif platform.system() == "Linux":
    cmd = asset_path("dssp-2.0.4-linux-amd64")
  else:
    raise Exception("Unsupported platform! Please try it under Linux.")

  pdbid = os.path.basename(pdbfn).replace(".pdb", "").upper()
  model = PDBParser(PERMISSIVE=1).get_structure(pdbid, pdbfn)[0]

  dsspfn = pdbfn.replace(".pdb", ".dssp")
  subprocess.check_call([cmd, '-i', pdbfn, '-o', dsspfn])

  dssp, keys = make_dssp_dict(dsspfn)
  idx, res, ss = [], [], []
  for k in keys:
    try:
      i, r, s, c = k[1][1], dssp[k][0], dssp[k][1], model[k[0]][k[1]]["CA"].get_coord()
      idx.append(i)
      res.append(r)
      ss.append(s)
    except KeyError: pass

  fafn = pdbfn.replace(".pdb", ".fa")
  with open(fafn, "w") as f:
    f.write(">%s\n" % pdbid)
    tmp = "".join(res)
    while tmp:
      if len(tmp) > 120:
        f.write("%s\n" % tmp[:120])
        tmp = tmp[120:]
      else:
        f.write("%s\n" % tmp)
        tmp = ""

def loadPDB(pdbfn, fraglen=4, mingap=0, mincont=2, maxdist=16.0):
  pdbid = os.path.basename(pdbfn).replace(".pdb", "").upper()
  model = PDBParser(PERMISSIVE=1).get_structure(pdbid, pdbfn)[0]

  dsspfn = pdbfn.replace(".pdb", ".dssp")
  if not os.path.isfile(dsspfn):
    convertPDB(pdbfn)

  dssp, keys = make_dssp_dict(dsspfn)

  idx, res, ss, coord = [], [], [], []
  for k in keys:
    try:
      i, r, s, c = k[1][1], dssp[k][0], dssp[k][1], model[k[0]][k[1]]["CA"].get_coord()
      idx.append(i)
      res.append(r)
      ss.append(s)
      coord.append(c)
    except KeyError: pass
  idx = np.array(idx)
  res = np.array(res)
  ss = np.array(ss)
  coord = np.array(coord)
  dist = squareform(pdist(coord))

  with TimeIt("gen data"):
    data = []
    possible_frag_j_idx = []
    possible_frag_i_idx = []
    for j in range(len(dist)-fraglen + 1):
      if not np.any(dist[j:j+fraglen, j:j+fraglen] >= maxdist):
        if(j >= fraglen+mingap):
          possible_frag_j_idx.append(j)
        possible_frag_i_idx.append(j)


    for i in possible_frag_i_idx:
      for j in possible_frag_j_idx:
        if i >= j-fraglen-mingap + 1:
          continue

        if np.any(dist[i:i+fraglen, j:j+fraglen] >= maxdist): continue
        if np.sum(dist[i:i+fraglen, j:j+fraglen] <= 8.0) < mincont: continue
        
        k = np.concatenate((idx[i:i+fraglen], idx[j:j+fraglen]), axis=0)
        r = np.concatenate((res[i:i+fraglen], res[j:j+fraglen]), axis=0)
        s = np.concatenate((ss[i:i+fraglen], ss[j:j+fraglen]), axis=0)
        c = np.concatenate((coord[i:i+fraglen], coord[j:j+fraglen]), axis=0)
        d0 = np.concatenate((dist[i:i+fraglen, i:i+fraglen], dist[i:i+fraglen, j:j+fraglen]), axis=1)
        d1 = np.concatenate((dist[j:j+fraglen, i:i+fraglen], dist[j:j+fraglen, j:j+fraglen]), axis=1)
        d = squareform(np.concatenate((d0, d1), axis=0))

        data.append([d, c, r, s, k])
  data = np.array(data)

  if len(data) > 0: return np.stack(data[:, 0]), np.stack(data[:, 1]), np.stack(data[:, 2]), np.stack(data[:, 3]), np.stack(data[:, 4]), len(res), len(data)
  else: return None, None, None, None, None, len(res), len(data)

def encode(ss):
  a, b, c = 0, 0, 0
  for i in ss:
    if i in "H": a += 1
    elif i in "E": b += 1
    else: c += 1
  if a >= 2 and a+c >=3: return "A%d" % a
  elif b >= 2 and b+c >=3: return "B%d" % b
  elif c >= 3: return "C%d" % c
  else: return "D0";

def filter(ss, lst):
  index = []
  fraglen = int(ss.shape[-1] / 2) # In Python3, fraglen is float due to division.
  for i in ss:
    code0 = encode(i[:fraglen])
    code1 = encode(i[-fraglen:])
    code = "".join(sorted([code0, code1]))
    index.append(code in lst)
  return index

