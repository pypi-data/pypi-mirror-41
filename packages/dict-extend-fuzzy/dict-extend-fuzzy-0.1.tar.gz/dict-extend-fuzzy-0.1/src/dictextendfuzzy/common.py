from difflib import get_close_matches

def transform(key):
    return str(key).lower()

def get_best(dict_, key, level):
    if key is None:
        key = ''

    mapping = { transform(k):k for k in dict_.keys() } 
    keys = mapping.keys()

    best_keys = get_close_matches(transform(key), keys, n=1, cutoff=level)
    if best_keys:
        return mapping.get(best_keys[0])
    return None
