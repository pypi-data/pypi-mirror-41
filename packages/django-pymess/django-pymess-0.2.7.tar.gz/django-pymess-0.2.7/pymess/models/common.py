from django.db.models import Manager
from django.contrib.contenttypes.models import ContentType

from pymess.utils import has_int_pk


class RelatedObjectManager(Manager):

    def create_from_related_object(self, related_object):
        return self.create(
            object_id_int=related_object.pk if has_int_pk(related_object) else None,
            object_id=related_object.pk,
            content_type=ContentType.objects.get_for_model(related_object)
        )

    def create_from_related_objects(self, *related_objects):
        return [
            self.create_from_related_object(related_object)
            for related_object in related_objects
        ]
