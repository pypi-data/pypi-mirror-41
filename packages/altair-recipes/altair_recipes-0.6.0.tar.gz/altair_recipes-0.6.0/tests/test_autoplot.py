import altair_recipes as ar
import numpy as np
import pandas as pd
from altair_recipes.common import viz_reg_test
from altair_recipes.display_altair import show_test

#' <h2>Autoplot</h2>

#' Autoplot is very easy to use but can produce a variety of charts that are reasonably appropriate for the data to be displayed.  Here is a longish sequence of examples of what autoplot will do with different combinations of up to three categorical or numerical variables and different data sizes


test_size = 5000


def rand_cat(x, n):
    return (
        pd.Series((x + np.random.normal(size=test_size) * n) + 77)
        .astype(int)
        .apply(chr)
    )


np.random.seed(seed=0)
x = np.random.normal(size=test_size)
y = np.random.normal(size=test_size) + x
z = np.random.normal(size=test_size) + y
data = pd.DataFrame(
    dict(
        x=x,
        x_cat=rand_cat(x, 1),
        y=y,
        y_cat=rand_cat(y, 0.5),
        z=z,
        z_cat=rand_cat(z, 0.5),
    )
)

#' The test code was generated by this code. So meta!


#
# numvars = ["x", "y", "z"]
# catvars = ["x_cat", "y_cat", "z_cat"]
# n = 0
# for nvars in range(1, 4):
#     for ncatvars in range(0, nvars + 1):
#         vars = catvars[:ncatvars] + numvars[ncatvars:nvars]
#         for nrows in [10, 50, 250, 1000, 5000]:
#             n = n + 1
#             print(
#                 """
# #'  <h3> Test autoplot #{n}</h3>
#
# @viz_reg_test
# def test_autoplot_{n}():
#     return ar.autoplot(data.head({nrows}), columns={vars})
#
# show_test(test_autoplot_{n})
# """.format(
#                     nrows=nrows, vars=vars, n=n
#                 )
#             )
#'  <h3> Test autoplot #1</h3>


@viz_reg_test
def test_autoplot_1():
    return ar.autoplot(data.head(10), columns=["x"])


show_test(test_autoplot_1)


#'  <h3> Test autoplot #2</h3>


@viz_reg_test
def test_autoplot_2():
    return ar.autoplot(data.head(50), columns=["x"])


show_test(test_autoplot_2)


#'  <h3> Test autoplot #3</h3>


@viz_reg_test
def test_autoplot_3():
    return ar.autoplot(data.head(250), columns=["x"])


show_test(test_autoplot_3)


#'  <h3> Test autoplot #4</h3>


@viz_reg_test
def test_autoplot_4():
    return ar.autoplot(data.head(1000), columns=["x"])


show_test(test_autoplot_4)


#'  <h3> Test autoplot #5</h3>


@viz_reg_test
def test_autoplot_5():
    return ar.autoplot(data.head(5000), columns=["x"])


show_test(test_autoplot_5)


#'  <h3> Test autoplot #6</h3>


@viz_reg_test
def test_autoplot_6():
    return ar.autoplot(data.head(10), columns=["x_cat"])


show_test(test_autoplot_6)


#'  <h3> Test autoplot #7</h3>


@viz_reg_test
def test_autoplot_7():
    return ar.autoplot(data.head(50), columns=["x_cat"])


show_test(test_autoplot_7)


#'  <h3> Test autoplot #8</h3>


@viz_reg_test
def test_autoplot_8():
    return ar.autoplot(data.head(250), columns=["x_cat"])


show_test(test_autoplot_8)


#'  <h3> Test autoplot #9</h3>


@viz_reg_test
def test_autoplot_9():
    return ar.autoplot(data.head(1000), columns=["x_cat"])


show_test(test_autoplot_9)


#'  <h3> Test autoplot #10</h3>


@viz_reg_test
def test_autoplot_10():
    return ar.autoplot(data.head(5000), columns=["x_cat"])


show_test(test_autoplot_10)


#'  <h3> Test autoplot #11</h3>


@viz_reg_test
def test_autoplot_11():
    return ar.autoplot(data.head(10), columns=["x", "y"])


show_test(test_autoplot_11)


#'  <h3> Test autoplot #12</h3>


@viz_reg_test
def test_autoplot_12():
    return ar.autoplot(data.head(50), columns=["x", "y"])


show_test(test_autoplot_12)


#'  <h3> Test autoplot #13</h3>


@viz_reg_test
def test_autoplot_13():
    return ar.autoplot(data.head(250), columns=["x", "y"])


show_test(test_autoplot_13)


#'  <h3> Test autoplot #14</h3>


@viz_reg_test
def test_autoplot_14():
    return ar.autoplot(data.head(1000), columns=["x", "y"])


show_test(test_autoplot_14)


#'  <h3> Test autoplot #15</h3>


@viz_reg_test
def test_autoplot_15():
    return ar.autoplot(data.head(5000), columns=["x", "y"])


show_test(test_autoplot_15)


#'  <h3> Test autoplot #16</h3>


@viz_reg_test
def test_autoplot_16():
    return ar.autoplot(data.head(10), columns=["x_cat", "y"])


show_test(test_autoplot_16)


#'  <h3> Test autoplot #17</h3>


@viz_reg_test
def test_autoplot_17():
    return ar.autoplot(data.head(50), columns=["x_cat", "y"])


show_test(test_autoplot_17)


#'  <h3> Test autoplot #18</h3>


@viz_reg_test
def test_autoplot_18():
    return ar.autoplot(data.head(250), columns=["x_cat", "y"])


show_test(test_autoplot_18)


#'  <h3> Test autoplot #19</h3>


@viz_reg_test
def test_autoplot_19():
    return ar.autoplot(data.head(1000), columns=["x_cat", "y"])


show_test(test_autoplot_19)


#'  <h3> Test autoplot #20</h3>


@viz_reg_test
def test_autoplot_20():
    return ar.autoplot(data.head(5000), columns=["x_cat", "y"])


show_test(test_autoplot_20)


#'  <h3> Test autoplot #21</h3>


@viz_reg_test
def test_autoplot_21():
    return ar.autoplot(data.head(10), columns=["x_cat", "y_cat"])


show_test(test_autoplot_21)


#'  <h3> Test autoplot #22</h3>


@viz_reg_test
def test_autoplot_22():
    return ar.autoplot(data.head(50), columns=["x_cat", "y_cat"])


show_test(test_autoplot_22)


#'  <h3> Test autoplot #23</h3>


@viz_reg_test
def test_autoplot_23():
    return ar.autoplot(data.head(250), columns=["x_cat", "y_cat"])


show_test(test_autoplot_23)


#'  <h3> Test autoplot #24</h3>


@viz_reg_test
def test_autoplot_24():
    return ar.autoplot(data.head(1000), columns=["x_cat", "y_cat"])


show_test(test_autoplot_24)


#'  <h3> Test autoplot #25</h3>


@viz_reg_test
def test_autoplot_25():
    return ar.autoplot(data.head(5000), columns=["x_cat", "y_cat"])


show_test(test_autoplot_25)


#'  <h3> Test autoplot #26</h3>


@viz_reg_test
def test_autoplot_26():
    return ar.autoplot(data.head(10), columns=["x", "y", "z"])


show_test(test_autoplot_26)


#'  <h3> Test autoplot #27</h3>


@viz_reg_test
def test_autoplot_27():
    return ar.autoplot(data.head(50), columns=["x", "y", "z"])


show_test(test_autoplot_27)


#'  <h3> Test autoplot #28</h3>


@viz_reg_test
def test_autoplot_28():
    return ar.autoplot(data.head(250), columns=["x", "y", "z"])


show_test(test_autoplot_28)


#'  <h3> Test autoplot #29</h3>


@viz_reg_test
def test_autoplot_29():
    return ar.autoplot(data.head(1000), columns=["x", "y", "z"])


show_test(test_autoplot_29)


#'  <h3> Test autoplot #30</h3>


@viz_reg_test
def test_autoplot_30():
    return ar.autoplot(data.head(5000), columns=["x", "y", "z"])


show_test(test_autoplot_30)


#'  <h3> Test autoplot #31</h3>


@viz_reg_test
def test_autoplot_31():
    return ar.autoplot(data.head(10), columns=["x_cat", "y", "z"])


show_test(test_autoplot_31)


#'  <h3> Test autoplot #32</h3>


@viz_reg_test
def test_autoplot_32():
    return ar.autoplot(data.head(50), columns=["x_cat", "y", "z"])


show_test(test_autoplot_32)


#'  <h3> Test autoplot #33</h3>


@viz_reg_test
def test_autoplot_33():
    return ar.autoplot(data.head(250), columns=["x_cat", "y", "z"])


show_test(test_autoplot_33)


#'  <h3> Test autoplot #34</h3>


@viz_reg_test
def test_autoplot_34():
    return ar.autoplot(data.head(1000), columns=["x_cat", "y", "z"])


show_test(test_autoplot_34)


#'  <h3> Test autoplot #35</h3>


@viz_reg_test
def test_autoplot_35():
    return ar.autoplot(data.head(5000), columns=["x_cat", "y", "z"])


show_test(test_autoplot_35)


#'  <h3> Test autoplot #36</h3>


@viz_reg_test
def test_autoplot_36():
    return ar.autoplot(data.head(10), columns=["x_cat", "y_cat", "z"])


show_test(test_autoplot_36)


#'  <h3> Test autoplot #37</h3>


@viz_reg_test
def test_autoplot_37():
    return ar.autoplot(data.head(50), columns=["x_cat", "y_cat", "z"])


show_test(test_autoplot_37)


#'  <h3> Test autoplot #38</h3>


@viz_reg_test
def test_autoplot_38():
    return ar.autoplot(data.head(250), columns=["x_cat", "y_cat", "z"])


show_test(test_autoplot_38)


#'  <h3> Test autoplot #39</h3>


@viz_reg_test
def test_autoplot_39():
    return ar.autoplot(data.head(1000), columns=["x_cat", "y_cat", "z"])


show_test(test_autoplot_39)


#'  <h3> Test autoplot #40</h3>


@viz_reg_test
def test_autoplot_40():
    return ar.autoplot(data.head(5000), columns=["x_cat", "y_cat", "z"])


show_test(test_autoplot_40)


#'  <h3> Test autoplot #41</h3>


@viz_reg_test
def test_autoplot_41():
    return ar.autoplot(data.head(10), columns=["x_cat", "y_cat", "z_cat"])


show_test(test_autoplot_41)


#'  <h3> Test autoplot #42</h3>


@viz_reg_test
def test_autoplot_42():
    return ar.autoplot(data.head(50), columns=["x_cat", "y_cat", "z_cat"])


show_test(test_autoplot_42)


#'  <h3> Test autoplot #43</h3>


@viz_reg_test
def test_autoplot_43():
    return ar.autoplot(data.head(250), columns=["x_cat", "y_cat", "z_cat"])


show_test(test_autoplot_43)


#'  <h3> Test autoplot #44</h3>


@viz_reg_test
def test_autoplot_44():
    return ar.autoplot(data.head(1000), columns=["x_cat", "y_cat", "z_cat"])


show_test(test_autoplot_44)


#'  <h3> Test autoplot #45</h3>


@viz_reg_test
def test_autoplot_45():
    return ar.autoplot(data.head(5000), columns=["x_cat", "y_cat", "z_cat"])


show_test(test_autoplot_45)


#' To finish three corner cases when there's no overlap with categorical vars only

w = pd.Series(range(10))

no_overlap_data = pd.DataFrame(
    dict(
        x=pd.concat([w, w - 3]).astype(str),
        y=pd.concat([w, w]).astype(str),
        z=pd.concat([w, w]).astype(str),
    )
)


@viz_reg_test
def test_autoplot_CCC():
    return ar.autoplot(no_overlap_data)


show_test(test_autoplot_CCC)


@viz_reg_test
def test_autoplot_CC():
    return ar.autoplot(no_overlap_data, columns=["x", "y"])


show_test(test_autoplot_CC)


@viz_reg_test
def test_autoplot_C():
    return ar.autoplot(no_overlap_data.head(10), columns=["x"])


show_test(test_autoplot_C)
