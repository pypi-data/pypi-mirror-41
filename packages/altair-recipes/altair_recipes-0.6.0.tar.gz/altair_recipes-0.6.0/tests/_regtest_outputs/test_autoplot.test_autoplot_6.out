{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.6.0.json",
  "config": {
    "view": {
      "height": 300,
      "width": 400
    }
  },
  "data": {
    "values": [
      {
        "x_cat": "Q"
      },
      {
        "x_cat": "M"
      },
      {
        "x_cat": "M"
      },
      {
        "x_cat": "O"
      },
      {
        "x_cat": "O"
      },
      {
        "x_cat": "M"
      },
      {
        "x_cat": "L"
      },
      {
        "x_cat": "L"
      },
      {
        "x_cat": "M"
      },
      {
        "x_cat": "L"
      }
    ]
  },
  "encoding": {
    "x": {
      "field": "x_cat",
      "type": "nominal"
    },
    "y": {
      "aggregate": "count",
      "type": "quantitative"
    }
  },
  "height": 600,
  "mark": "bar",
  "width": 800
}